build/Targets/RepoToolset/SymStore.targets

```diff
@@ -0,0 +1,106 @@
+             Properties="TargetFramework=%(_TargetFramework.Identity)">
+    </MSBuild>
+  </Target>
+
+  <Target Name="_InnerDeployToSymStore">
+    <PropertyGroup>
+      <_TargetPdbPath />
+      <_TargetPdbPath Condition="'$(DebugType)' != 'embedded'">$([System.IO.Path]::ChangeExtension($(TargetPath), '.pdb'))</_TargetPdbPath>
+
+      <_SymStoreOutputDir>$(ArtifactsSymStoreDirectory)$(MSBuildProjectName)\$(TargetFramework)\</_SymStoreOutputDir>
+      <_SymStorePdbPath>$(_SymStoreOutputDir)$(TargetName).pdb</_SymStorePdbPath>
+      <_SymStoreAssemblyPath>$(_SymStoreOutputDir)$(TargetName)$(TargetExt)</_SymStoreAssemblyPath>
+      
+      <PublishOutputToSymStore Condition="'$(PublishOutputToSymStore)' == '' and Exists('$(TargetPath)') and ('$(DebugType)' == 'embedded' or Exists('$(_TargetPdbPath)'))">true</PublishOutputToSymStore>
+    </PropertyGroup>
+  </Target>
+
+  <!-- 
+    Convert Portable and Embedded PDBs. 
+    We publish converted PDBs to Symbol Store so that WinDBG can find them when debugging crash dumps.
+    Convert source link to srcsvr so that both WinDBG and VS can open the source files.
+  -->
+  <Target Name="_DeployPortableSymbolsToSymStore"
+          AfterTargets="_InnerDeployToSymStore"
+          Inputs="$(TargetPath);$(_TargetPdbPath)"
+          Outputs="$(_SymStorePdbPath)"
+          Condition="'$(PublishOutputToSymStore)' == 'true' and ('$(DebugType)' == 'portable' or '$(DebugType)' == 'embedded')">
+
+    <PropertyGroup>
+      <_PdbConverterPath>$(NuGetPackageRoot)pdb2pdb\$(Pdb2PdbVersion)\tools\Pdb2Pdb.exe</_PdbConverterPath>
+      <_PdbConverterCommandLineArgs>"$(TargetPath)" /out "$(_SymStorePdbPath)" /verbose</_PdbConverterCommandLineArgs>
+    </PropertyGroup>
+
+    <Error Text="Attempt to publish Portable PDB to SymStore without conversion: UsingToolPdbConverter property is false in project $(MSBuildProjectName)"
+           Condition="'$(UsingToolPdbConverter)' != 'true'"
+           ContinueOnError="false"/>
+
+    <MakeDir Directories="$(_SymStoreOutputDir)"/>
+   
+    <Exec Command='"$(_PdbConverterPath)" $(_PdbConverterCommandLineArgs)' IgnoreExitCode="false" />
+
+    <ItemGroup>
+      <FileWrites Include="$(_SymStorePdbPath)"/>
+    </ItemGroup>
+  </Target>
+
+  <!-- 
+    The current MicroBuild symbol indexing task doesn't support specifying list of files to index.
+    Therefore we create hardlinks to the files in output directory, which will get signed.
+  -->
+  <Target Name="_DeployWindowsSymbolsToSymStore"
+          AfterTargets="_InnerDeployToSymStore"
+          Condition="'$(PublishOutputToSymStore)' == 'true' and ('$(DebugType)' == 'full' or '$(DebugType)' == 'pdbonly')"
+          Inputs="$(_TargetPdbPath)"
+          Outputs="$(_SymStorePdbPath)">
+
+    <MakeDir Directories="$(_SymStoreOutputDir)"/>
+  
+    <Copy SourceFiles="$(_TargetPdbPath)"
+          DestinationFiles="$(_SymStorePdbPath)"
+          UseHardlinksIfPossible="true"/>
```

Comment:
Are we actually relying on the hardlinks for correctness here? The comment implies we're expecting to be able to modify one and the other gets up to date.

Reply:
We are not. I ordered the targets so that it works even if this makes a copy instead of a hard link. Will add a comment.

---
In reply to: [147293638](https://github.com/dotnet/roslyn/pull/22850#discussion_r147293638) [](ancestors = 147293638)